pulp_container/app/tasks/sign.py

```diff
@@ -0,0 +1,122 @@
+import base64
+import hashlib
+import os
+import tempfile
+
+from django.conf import settings
+
+from pulpcore.plugin.models import Repository
+
+from pulp_container.app.models import (
+    ManifestSignature,
+    ManifestSigningService,
+    Tag,
+)
+
+from pulp_container.app.utils import extract_data_from_signature
+from pulp_container.constants import (
+    MANIFEST_MEDIA_TYPES,
+    SIGNATURE_TYPE,
+)
+
+
+def sign(repository_pk, signing_service_pk, reference, tags_list=None):
+    """
+    Create a new repository version by signing manifests.
+
+    Create signature for each manifest that is specified and add it to the repo.
+    If no manifests were specified then sign all manifests in the repo.
+
+    What manifests to sign is identified by tag.
+    Manifests list are not signed. Image manifests from manifest list are signed by digest/
+
+    Args:
+        repository_pk (int): The pk for a Repository for which a new Repository Version should be
+                             created.
+        signing_service_pk (int): pk of the signing service to use.
+        reference (str): Reference that will be used to produce signature.
+        tags_list (list): List of PKs for :class:`~pulp_container.app.models.Tag` manifests of which
+                          should be signed.
+
+    """
+    repository = Repository.objects.get(pk=repository_pk).cast()
+    latest_version = repository.latest_version()
+    if tags_list:
+        latest_repo_tags = latest_version.content.filter(
+            pulp_type=Tag.get_pulp_type(), pk__in=tags_list
+        )
```

Comment:
I think I followed the copy logic, where we do not validate provided list of content but copy only what is found. I would call on get_resource in case we'd switch to list of hrefs instead, but otherwise I am not sure this is worth the hassle.
We could look into adding such validation but then we should do this across other tasks a swell.